app.py

```diff
@@ -0,0 +1,81 @@
+from flask import Flask, render_template, request, redirect, url_for, flash, session
+from flask_sqlalchemy import SQLAlchemy
+from flask_bcrypt import Bcrypt
+from config import Config
+from models import db, User
+
+app = Flask(__name__)
+app.config.from_object(Config)
+
+# Initialize extensions
+db.init_app(app)
+bcrypt = Bcrypt(app)
+
+# Create tables explicitly during app initialization
+with app.app_context():
+    db.create_all()
+
+# Routes
+@app.route('/')
+def home():
+    return render_template('index.html')
+
+@app.route('/register', methods=['GET', 'POST'])
+def register():
+    if request.method == 'POST':
+        username = request.form['username']
+        email = request.form['email']
+        password = request.form['password']
+
+        if User.query.filter_by(email=email).first():
+            flash('Email already registered!', 'danger')
+            return redirect(url_for('register'))
+
+        if User.query.filter_by(username=username).first():
+            flash('Username already taken!', 'danger')
```

Comment:
Change the error message. We don't want to introduce `username enumeration owasp` in our system.
Referring to following issue that I have created:
https://github.com/mehedinaeem/PROJECT_/issues/1

Add similar descriptions for your issue as well. Or you can just copy paste the requirements to your existing issue.